sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -103,7 +108,8 @@ private static bool IsReturningCollection(SyntaxNodeAnalysisContext context)
 
             return methodSymbol != null &&
                 !methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
+                methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes) &&
+                !methodSymbol.ReturnType.DerivesOrImplementsAny(IgnoredTypes);
```

Comment:
[minor] I would use only `DerivesFromAny` to avoid the extra check if implements when it doesn't derive as for now the collection contains only a class.